crates/networking/rpc/clients/eth/mod.rs

```diff
@@ -169,6 +169,21 @@ impl EthClient {
         response
     }
 
+    async fn send_request_to_all(
+        &self,
+        request: RpcRequest,
+    ) -> Result<RpcResponse, EthClientError> {
+        let mut response = Err(EthClientError::Custom("All rpc calls failed".to_string()));
+
+        for url in self.urls.iter() {
+            let maybe_response = self.send_request_to_url(url, &request).await;
+            if maybe_response.is_ok() {
+                response = maybe_response;
+            }
+        }
+        response
```

Comment:
Think this could save some waiting times
```suggestion
        let mut handlers = Vec::with_capacity(self.urls.len());

        for url in self.urls.iter() {
            handlers.push(self.send_request_to_url(url, &request));
        }

        for handler in handlers {
            if let Ok(res) = handler.await {
                response = Ok(res);
            }
        }

        response
```